compiler/rustc_parse/src/parser/mod.rs

```diff
@@ -347,7 +347,7 @@ impl TokenCursor {
                         .cloned()
                         .collect::<TokenStream>()
                 },
-                false,
+                true,
```

Comment:
Any assumptions like this make fixing https://github.com/rust-lang/rust/issues/67062 harder.

We should drop `NoDelim` delimiters as late as possible, in a single place, and with a large FIXME.

Reply:
https://github.com/rust-lang/rust/pull/96210#discussion_r854037894